src/prefect/storage/git.py

```diff
@@ -0,0 +1,149 @@
+            for common git hosting sites
+        - **kwargs (Any, optional): any additional `Storage` initialization options
+    """
+
+    def __init__(
+        self,
+        flow_path: str,
+        repo: str,
+        repo_host: str = "github.com",
+        flow_name: str = None,
+        git_token_secret_name: str = None,
+        git_token_username: str = None,
+        branch_name: str = None,
+        tag: str = None,
+        clone_depth: int = 1,
+        use_ssh: bool = False,
+        format_access_token: bool = True,
+        **kwargs: Any,
+    ) -> None:
+        if tag and branch_name:
+            raise ValueError(
+                "Either `tag` or `branch_name` can be specified, but not both"
+            )
+
+        self.flow_path = flow_path
+        self.repo = repo
+        self.repo_host = repo_host
+        self.flow_name = flow_name
+        self.git_token_secret_name = git_token_secret_name
+        self.git_token_username = (
+            git_token_username if git_token_username else repo.split("/")[0]
+        )
+        self.branch_name = branch_name
+        self.tag = tag
+        self.clone_depth = clone_depth
+        self.use_ssh = use_ssh
+        self.format_access_token = format_access_token
+        super().__init__(**kwargs)
+
+    def get_flow(self, flow_name: str) -> "Flow":
+        """
+        Given a flow name within this Storage object, load and return the Flow.
+        Args:
+            - flow_name (str): the name of the flow to return.
+        Returns:
+            - Flow: the requested flow
+        """
+        if flow_name not in self.flows:
+            raise ValueError("Flow is not contained in this Storage")
+
+        with TemporaryGitRepo(
+            git_clone_url=self.git_clone_url,
+            branch_name=self.branch_name,
+            tag=self.tag,
+            clone_depth=self.clone_depth,
+        ) as temp_repo:
+            flow = extract_flow_from_file(
+                file_path=os.path.join(temp_repo.temp_dir.name, self.flow_path),
+                flow_name=self.flow_name,
+            )
+        return flow
+
+    def add_flow(self, flow: "Flow") -> str:
+        """
+        Method for storing a new flow as bytes in the local filesytem.
+        Args:
+            - flow (Flow): a Prefect Flow to add
+        Returns:
+            - str: the location of the added flow in the repo
+        Raises:
+            - ValueError: if a flow with the same name is already contained in this storage
+        """
+        if flow.name in self:
+            raise ValueError(
+                'Name conflict: Flow with the name "{}" is already present in this storage.'.format(
+                    flow.name
+                )
+            )
+        self.flows[flow.name] = self.flow_path
+        self._flows[flow.name] = flow
+        return self.flow_path
+
+    @property
+    def git_token_secret(self) -> str:
```

Comment:
Should probably support loading from an environment variable too so they don't need to use Prefect secrets to set this

Reply:
Hmm are there other good examples of how we'd want this pattern to work?

My cheeky response is, "just set the environment variable `export PREFECT__CONTEXT__SECRETS__MYSECRET="MY SECRET VALUE"`:".

Reply:
Idk how strongly I feel but the pattern used in all the other storage's is like so
```python
        if self.access_token_secret is not None:
            # If access token secret specified, load it
            access_token = Secret(self.access_token_secret).get()
        else:
            # Otherwise, fallback to loading from local secret or environment variable
            access_token = prefect.context.get("secrets", {}).get("GITHUB_ACCESS_TOKEN")
            if access_token is None:
                access_token = os.getenv("GITHUB_ACCESS_TOKEN")
 ```

cc @cicdw @dylanbhughes on continuing to support this pattern

Reply:
I would vote to stop supporting this pattern

Reply:
I defer to @cicdw. My feedback is that the user shouldn't have to stick with a particular name for a Prefect secret so that a single tenant can have more than one secret for GitHub storage and that's well-documented. As long as that's achieved I'm 👍